babel/dates.py

```diff
@@ -1161,7 +1165,12 @@ def parse_date(string, locale=LC_TIME, format='medium'):
     if not numbers:
         raise ParseError("No numbers were found in input")
 
-    # TODO: try ISO format first?
+    # we try ISO-alike format first, meaning similar to YYYY-MM-DD
+    if len(numbers) == 3:
```

Comment:
I think we should at the very least check that the first number-string is 4-digited:
```suggestion
    if len(numbers) == 3 and len(numbers[0]) == 4:
```
or maybe even more specifically
```suggestion
    if [len(n) for n in numbers] == [4, 2, 2]:  # 3 numbers, "ISO-shaped"
```
...

Reply:
I replaced the whole thing through a single pattern matching, and else let datetime.date function do the heavy lifting validation. This is shorter, more expressive, and allow me to support as well the basic YYYYMMDD format of ISO.